### PR TITLE
feat(proxies): use pypac to use PAC file

### PR DESCRIPTION
**Pull request by jmkerloch**
Introduce support for PAC file for proxy definition.

QDT will first try to use `QDT_PROXY_HTTP` environment variable then new environment variable `QDT_PAC_FILE` for custom PAC file then system PAC if available.

### CODE REVIEW

File: docs/guides/howto_behind_proxy.md [hunk 1]
```diff
@@ -28,6 +28,14 @@ qdt --proxy-http "http://user:password@proxyserver.intra:8765"
 - it avoids potential conflict with "classic" proxy settings
 - it allows to use a specific network proxy for QDT (can be useful for some well controlled systems)
 
+### Use PAC file
+
+[PAC file](https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling/Proxy_Auto-Configuration_PAC_file) can be used by SysAdmin to define proxy with a set of rules depending on the url.
+
+For QDT, PAC file can be used if no environment variable for proxy are already defined (`HTTP_PROXY`,`HTTPS_PROXY`, `QDT_PROXY_HTTP`).
```
Comment (Guts) [inline docs/guides/howto_behind_proxy.md:35]: Change priority:

1. QDT_PROXY_HTTP
2. System settings ---> pac or raw defined
3. HTTP_PROXY

File: docs/guides/howto_behind_proxy.md [hunk 2]
```diff
@@ -28,6 +28,14 @@ qdt --proxy-http "http://user:password@proxyserver.intra:8765"
...
+
+[PyPac](https://pypac.readthedocs.io/en/latest/) is used for PAC file management. By default we are using the PAC file defined by system but a custom PAC file can be defined with `QDT_PAC_FILE` environment variable (local file or url).
```
Comment (Guts) [inline docs/guides/howto_behind_proxy.md:37]: Add this environment variable to settings page.

File: qgis_deployment_toolbelt/utils/proxies.py [hunk 3]
```diff
@@ -31,13 +35,15 @@
 # ########## Functions #############
 # ##################################
 @lru_cache
-def get_proxy_settings() -> dict:
+def get_proxy_settings(url: str | None = None) -> dict:
     """Retrieves network proxy settings from operating system configuration or
     environment variables.
-
+    Args:
+        url (str): url for request in case of PAC file use
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/proxies.py:42]: Inconsistent docstring. Missing optional mention, no?

File: requirements/base.txt [hunk 4]
```diff
@@ -9,3 +9,4 @@ pyyaml>=5.4,<7
 pywin32==306 ; sys_platform == 'win32'
 requests>=2.31,<3
 truststore>=0.8,<1
+pypac>=0.16.3,<1
```
Comment (Guts) [inline requirements/base.txt:12]: requirements should be A-z ordered

File: tests/fixtures/pac/proxy.pac
Comment (Guts) [inline tests/fixtures/pac/proxy.pac:1]: Hum, not sure that we're allowed to broadcast this file. I suggest to remove it and clean up the git history also if possible.

**Review by Guts — CHANGES_REQUESTED** [review]

File: docs/guides/howto_behind_proxy.md (hunk 1 above)
Comment (jmkerloch) [reply docs/guides/howto_behind_proxy.md]: This was updated.

File: tests/fixtures/pac/proxy.pac
Comment (jmkerloch) [reply tests/fixtures/pac/proxy.pac:1]: File removed and git history also

**Review by Guts — APPROVED** [review]

**Pull request merged**